app/lib/features/pins/pages/pins_list_page.dart

```diff
@@ -0,0 +1,188 @@
+    hasSearchTerm = searchValue.isNotEmpty;
+
+    if (hasSearchTerm) {
+      pinList = ref.watch(
+        pinListSearchProvider(
+          (spaceId: widget.spaceId, searchText: searchValue),
+        ),
+      );
+    } else {
+      pinList = ref.watch(pinListProvider(widget.spaceId));
+    }
+
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        _buildSearchBar(),
+        Expanded(
+          child: pinList.when(
+            data: (pins) => _buildPinsList(pins),
+            error: (error, stack) =>
+                Center(child: Text(L10n.of(context).loadingFailed(error))),
+            loading: () => Center(child: Text(L10n.of(context).loading)),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Widget _buildSearchBar() {
+    return Container(
+      margin: const EdgeInsets.symmetric(horizontal: 10, vertical: 15),
+      child: SearchBar(
+        controller: searchTextController,
+        leading: const Padding(
+          padding: EdgeInsets.all(8.0),
+          child: Icon(Atlas.magnifying_glass),
+        ),
+        hintText: L10n.of(context).search,
+        trailing: hasSearchTerm
+            ? [
+                IconButton(
+                  onPressed: () {
+                    FocusManager.instance.primaryFocus?.unfocus();
+                    ref.read(searchValueProvider.notifier).state = '';
+                    searchTextController.clear();
+                  },
+                  icon: const Icon(Icons.clear),
+                ),
+              ]
+            : null,
+        onChanged: (value) =>
+            ref.read(searchValueProvider.notifier).state = value,
+      ),
+    );
+  }
+
+  Widget _buildPinsList(List<ActerPin> pins) {
+    final size = MediaQuery.of(context).size;
+    final widthCount = (size.width ~/ 500).toInt();
+    const int minCount = 2;
+
+    if (pins.isEmpty) return _buildPinsEmptyState();
+
+    return SingleChildScrollView(
+      child: StaggeredGrid.count(
+        crossAxisCount: max(1, min(widthCount, minCount)),
+        children: [
+          for (var pin in pins)
+            PinListItemById(
+              pinId: pin.eventIdStr(),
+              showSpace: widget.spaceId == null,
+            ),
+        ],
+      ),
+    );
+  }
+
+  Widget _buildPinsEmptyState() {
+    bool canAdd = true;
+    if (widget.spaceId != null && !hasSearchTerm) {
+      final membership = ref.watch(roomMembershipProvider(widget.spaceId!));
+      canAdd = membership.requireValue!.canString('CanPostPin');
+    }
+    return Center(
+      heightFactor: 1,
+      child: EmptyState(
+        title: hasSearchTerm
+            ? L10n.of(context).noMatchingPinsFound
+            : L10n.of(context).noPinsAvailableYet,
+        subtitle: L10n.of(context).noPinsAvailableDescription,
+        image: 'assets/images/empty_pin.svg',
+        primaryButton: canAdd && !hasSearchTerm
+            ? ActerPrimaryActionButton(
```

Comment:
should use the same `hasSpaceWithPermissionProvider`

Reply:
Sure